### PR TITLE
feat(protocol-designer): add export warning modal

### PR DESCRIPTION
**Pull request by ncdiehl11**
# Overview

If a user exports a protocol that doesn't use some hardware (instruments, modules, staging areas, trash entities), we warn them on export with an itemized list of any implicated hardware. Here, I update the utility for getting content for this modal depending on the specific unused hardware. I also wire up the modal using our `useBlockingHint` hook in `ProtocolOverview`

Closes AUTH-818

## Test Plan and Hands on Testing

- create a protocol with no steps
- export and verify that "protocol has no steps" warning modal shows
- exit the modal and add steps and some hardware that is not used in the protocol (i.e., gripper without using a move labware, heater shaker without adding any heater-shaker steps, etc.).
- export again
- verify that 'protocol has unused hardware' warning modal shows
- check box to not show again and export
- start a new session and import the protocol that you just exported
- select "export" and verify that the modal does not show

## Changelog

- implement `useBlockingHint` in `ProtocolOverview` on export
- expand `HintKey` type to include new title keys for export warning modal
- add new translations 
- update tests

## Review requests

see test plan. Please verify that hint logic is correct

## Risk assessment

low

### CODE REVIEW

File: protocol-designer/src/organisms/BlockingHintModal/index.tsx [hunk 1]
```diff
@@ -38,7 +38,6 @@ export function BlockingHintModal(props: HintProps): JSX.Element {
   }, [])
 
   const onCancelClick = (): void => {
-    dispatch(actions.removeHint(hintKey, rememberDismissal))
```
Comment (jerader) [inline protocol-designer/src/organisms/BlockingHintModal/index.tsx:41]: oof good catch!

File: protocol-designer/src/pages/ProtocolOverview/__tests__/ProtocolOverview.test.tsx [hunk 2]
```diff
@@ -21,10 +21,12 @@ import { LiquidDefinitions } from '../LiquidDefinitions'
 import { StepsInfo } from '../StepsInfo'
 
 import type { NavigateFunction } from 'react-router-dom'
+import { getDismissedHints } from '../../../tutorial/selectors'
```
Comment (jerader) [inline protocol-designer/src/pages/ProtocolOverview/__tests__/ProtocolOverview.test.tsx]: import can go higher

**Review by jerader — COMMENTED** [review]
looking good! only thing i noticed is it doesn't account for multiples of a module? I added a bunch of stuff including 2 temperature modules, only shows one

Screenshot 2024-10-29 at 13 36 29

**Comment by jerader** [comment]
also small nit but your pr doesn't seem to touch redux anymore, so do you mind removing it from the title?

**Comment by ncdiehl11** [comment]
> looking good! only thing i noticed is it doesn't account for multiples of a module? I added a bunch of stuff including 2 temperature modules, only shows one
> 
> Screenshot 2024-10-29 at 13 36 29

Do you mind sending your protocol over? I am seeing both modules
Screenshot 2024-10-29 at 2 42 36 PM

File: protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/PauseTools/index.tsx [hunk 3]
```diff
@@ -82,6 +83,15 @@ export function PauseTools(props: StepFormProps): JSX.Element {
 
   const { pauseAction } = props.formData
 
+  useEffect(() => {
+    if (
+      pauseAction === PAUSE_UNTIL_RESUME ||
+      pauseAction === PAUSE_UNTIL_TIME
+    ) {
+      propsForFields.moduleId.updateValue(null)
```
Comment (jerader) [inline protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/PauseTools/index.tsx]: why do we need to set the moduleId as null? is not null by default?

**Review by jerader — APPROVED** [review]
lgtm! just left a question about the `pauseTools` change

File: protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/PauseTools/index.tsx (hunk 3 above)
Comment (ncdiehl11) [reply protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/PauseTools/index.tsx:91]: Sorry, forgot to add context here. The issue is when you select 'pause until temp', select a module from the dropdown, then switch to a different pause action and save. the module data was still saved in the form even though it is not used. this resulted in `getUnusedEntities` ingesting step forms in which the module is seemingly used even though it was not in reality

Comment (ncdiehl11) [reply protocol-designer/src/pages/Designer/ProtocolSteps/StepForm/StepTools/PauseTools/index.tsx]: good catch, I just wired this up in dependentFieldsUpdatePause instead

**Pull request merged**